src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -210,32 +253,52 @@ public async Task ConfigureDeployment(CloudApplication cloudApplication, Orchest
             }
         }
 
-        private async Task<List<Recommendation>> GenerateDeploymentRecommendations(Orchestrator orchestrator)
+        private async Task<List<Recommendation>> GenerateDeploymentRecommendations(Orchestrator orchestrator, string deploymentProjectPath)
         {
-            var recommendations = await orchestrator.GenerateDeploymentRecommendations();
-            if (recommendations.Count == 0)
+            List<Recommendation> recommendations;
+            if (!string.IsNullOrEmpty(deploymentProjectPath))
             {
-                throw new FailedToGenerateAnyRecommendations("The project you are trying to deploy is currently not supported.");
+                recommendations = await orchestrator.GenerateRecommendationsFromSavedDeploymentProject(deploymentProjectPath);
+                if (recommendations.Count == 0)
+                {
+                    var errorMessage = $"Could not find any deployment recipe located inside '{deploymentProjectPath}' that can be used for deployment of the target application";
+                    throw new FailedToGenerateAnyRecommendations(errorMessage);
+                }
+            }
+            else
+            {
+                recommendations = await orchestrator.GenerateDeploymentRecommendations();
+                if (recommendations.Count == 0)
+                {
+                    var errorMessage = "The project you are trying to deploy is currently not supported.";

```

Comment:
Suggested text:

"There are no compatible deployment recommendations for this application."